src/sentry/seer/similarity/utils.py

```diff
@@ -224,18 +225,14 @@ def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
         frame_count += len(contributing_frames)
 
         for frame in contributing_frames:
-            frame_dict = {"filename": "", "function": "", "context-line": "", "module": ""}
-            for frame_values in frame.get("values", []):
-                if frame_values.get("id") in frame_dict:
-                    frame_dict[frame_values["id"]] = _get_value_if_exists(frame_values)
+            frame_dict = extract_values_from_frame_values(frame.get("values", []))
+            filename = extract_filename(frame_dict)
```

Comment:
It will grab the filename from `filename` or `module`.
<img width="660" alt="image" src="https://github.com/user-attachments/assets/f63cdbf4-0e29-4f56-a1d3-2b30d9ac7a63">